route.go

```diff
@@ -396,7 +396,7 @@ func (r *Route) Schemes(schemes ...string) *Route {
 	for k, v := range schemes {
 		schemes[k] = strings.ToLower(v)
 	}
-	if r.buildScheme == "" && len(schemes) > 0 {
+	if r.getBuildScheme() == "" && len(schemes) > 0 {
```

Comment:
My thinking was that since the choice was made that building a URL uses the first scheme set, the extension of that for subrouters would be using the first scheme found starting at the root.

I am just now starting to understand the relationship between routes with schemes at multiple levels. When matching, I suppose the parent would need to include all schemes that its child routes care about. The children could either match all schemes of the parent, or a subset.

Given that relationship, I see how building a URL for a child route should prefer the scheme of the child (if present), rather than the way I implemented this.

Do you agree? If so, I can add a test to verify that behavior and update the implementation.